xlators/features/simple-quota/src/simple-quota.c

```diff
@@ -0,0 +1,1295 @@
+    GF_OPTION_RECONF("pass-through", this->pass_through, options, bool, out);
+    GF_OPTION_RECONF("use-backend", priv->use_backend, options, bool, out);
+
+out:
+    return 0;
+}
+
+int32_t
+init(xlator_t *this)
+{
+    sq_private_t *priv;
+    int ret = -1;
+
+    if (!this->children || this->children->next) {
+        gf_log(this->name, GF_LOG_ERROR,
+               "FATAL: simple-quota should have exactly one child");
+        goto out;
+    }
+
+    if (!this->parents) {
+        gf_log(this->name, GF_LOG_WARNING, "dangling volume. check volfile ");
+    }
+
+    priv = GF_CALLOC(sizeof(sq_private_t), 1, gf_common_mt_char);
+    if (!priv)
+        goto out;
+
+    GF_OPTION_INIT("pass-through", this->pass_through, bool, out);
+    GF_OPTION_INIT("use-backend", priv->use_backend, bool, out);
+    GF_OPTION_INIT("cmd-from-all-client", priv->take_cmd_from_all_client, bool,
+                   out);
+
+    /* By default assume this is true, if there is a setxattr to set
+       the total usage, then mark it false  */
+    priv->allow_fops = false;
+    priv->no_distribute = true;
+    INIT_LIST_HEAD(&priv->ns_list);
+    LOCK_INIT(&priv->lock);
+    this->private = priv;
+    // quota_set_thread(this);
+
+    gf_log(this->name, GF_LOG_DEBUG, "Simple Quota xlator loaded");
+    ret = 0;
+
+out:
+    return ret;
+}
+
+void
+fini(xlator_t *this)
+{
+    sq_private_t *priv = this->private;
+
+    if (!priv)
+        return;
+
+    // sq_clear_thread(this, priv->quota_set_thread);
+    gf_log(this->name, GF_LOG_TRACE, "calling fini, nothing major pending");
+    this->private = NULL;
+    GF_FREE(priv);
+
+    return;
+}
+
+int
+notify(xlator_t *this, int32_t event, void *data, ...)
+{
+    if (GF_EVENT_PARENT_DOWN == event) {
+        gf_log(this->name, GF_LOG_DEBUG,
+               "sending all pending information to disk");
+        sync_data_from_priv(this, this->private);
+    }
+
+    return default_notify(this, event, data);
+}
+
+struct xlator_cbks cbks = {
+    .forget = sq_forget,
+};
+
+struct xlator_fops fops = {
+    /* Very critical fop */
+    .statfs = sq_statfs,
+
+    /* Required to handle hardlimit etc */
+    .setxattr = sq_setxattr,
+
+    /* Set the inode context with limit etc during first lookup */
+    .lookup = sq_lookup,
+
+    /* Implement a check for usage */
+    .writev = sq_writev,
+    .create = sq_create,
+    .mkdir = sq_mkdir,
+
+    /* use for update */
+    .unlink = sq_unlink,
+    .rmdir = sq_rmdir,
+    .truncate =
+        sq_truncate, /* not implementing a check as it would punch hole */
+    .ftruncate = sq_ftruncate,
+    .fallocate = sq_fallocate,
+    .discard = sq_discard,
+
+    /* rename should reduce the space used of target entry if present */
+    .rename = sq_rename,
+};
+
+struct volume_options options[] = {
+    {
+        .key = {"pass-through"},
+        .type = GF_OPTION_TYPE_BOOL,
+        .default_value = "true",
+        .op_version = {GD_OP_VERSION_9_0},
```

Comment:
Should here be use GD_ OP_ VERSION_ 11_ 0？

Reply:
Yes. Thanks for pointing. Been more than a year since first coding this, and hence the issue :-) 

Will fix.